crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -117,6 +118,78 @@ impl ArrayHeap {
         self.staged.insert(index, (value, op));
     }
 
+    /// This helper function transforms an expression into a (univariate) affine expression, by generating a witness if the input expression is not linear nor constant
+    fn normalize_expression(expr: &Expression, evaluator: &mut Evaluator) -> Expression {
+        if expr.is_linear() && expr.linear_combinations.len() <= 1 {
+            expr.clone()
+        } else {
+            let w = evaluator.create_intermediate_variable(expr.clone());
+            Expression::from(w)
+        }
+    }
+
+    /// Decide which opcode to use, depending on the backend support
+    fn acir_gen(
+        &self,
+        evaluator: &mut Evaluator,
+        array_id: ArrayId,
+        array_len: u32,
+        is_opcode_supported: IsOpcodeSupported,
+    ) {
+        let dummy = MemoryBlock { id: AcirBlockId(0), len: 0, trace: Vec::new() };
+
+        if is_opcode_supported(&AcirOpcode::ROM(dummy.clone())) {
```

Comment:
Using `is_opcode_supported` here seems a bit off since we also have a fallback mechanism in acvm [here](https://github.com/noir-lang/acvm/blob/efd37fedcbbabb3fac810e662731439e07fef49a/acvm/src/compiler.rs#L36) and then we also have one in the frontend.

Why can't the fallback be put into ACVM?